### PR TITLE
Gym Networking

### PR DESCRIPTION
**Pull request by jmf373**
- Refactored the Gym model to make much more sense
- Integrated new Gym model
- Created a query to fetch all gyms

### CODE REVIEW

File: Fitness/Extensions/Date+Shared.swift [hunk 1]
```diff
@@ -34,13 +34,24 @@ extension Date {
 
         return calendar.date(from: dateComponents)!
     }
+    
+    static public func getDatetimeFromString(datetime: String?) -> Date {
+        guard let datetime = datetime else {
```
[the remark this answers is not in the document]
Comment (jhseung) [reply Fitness/Extensions/Date+Shared.swift]: This is more of the swift-y way of doing it imo

**Comment by young-k** [comment]
Another comment –– for PRs can we have this just be "Gym Networking", there's no need to put the name in the PR title

File: Fitness/Extensions/Date+Shared.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmf373) [reply Fitness/Extensions/Date+Shared.swift]: I don't think `datetime else return Date()` is valid syntax lol

**Comment by jmf373** [comment]
@jhseung Yup I was planning on just replacing the functions in `NetworkManager`, although some won't be needed anymore because the models are structured better and graphQL simplifies some things

**Pull request merged**